godot-core/src/init/mod.rs

```diff
@@ -34,6 +34,17 @@ pub unsafe fn __gdext_load_library<E: ExtensionLibrary>(
             Some(false) => sys::linux_reload_workaround::disable_hot_reload(),
         }
 
+        #[cfg(not(wasm_nothreads))]
+        if crate::meta::MAIN_THREAD_ID
+            .set(std::thread::current().id())
+            .is_err()
+        {
+            eprintln!("Failed to set MAIN_THREAD_ID during initialization!");
+
+            // Return initialization failure.
+            return 0;
+        };
```

Comment:
Maybe write an explicit comment that this comes after the hot-reload workaround, to avoid future temptation to move it up.

Also, you can use return `SYS_FALSE` instead of `0`:
https://github.com/godot-rust/gdext/blob/cadfd16b59e14f9365e2b16cbbacb829e2e7c8ca/godot-ffi/src/conv.rs#L78-L79

Reply:
Disappears entirely with `ManualInitCell<T>`.